testcases/kernel/syscalls/madvise/madvise08.c

```diff
@@ -41,8 +41,11 @@
 
 static int dfd;
 static void *fmem;
-static char cpattern[CORENAME_MAX_SIZE];
-static int restore_cpattern;
+
+static const char * const save_restore[] = {
+	CORE_PATTERN,
+	NULL,
+};
 
```

Comment:
I would put this structure just before the tst_test structure or even embed it there as:

```
static struct tst_test test = {
...
   .save_restore = (const char* const[]) {
       CORE_PATTERN,
       NULL
   }
};

Reply:
I used ltp/lib/newlib_tests/test19.c as a reference